lib/jekyll-linkpreview.rb

```diff
@@ -145,6 +146,7 @@ def initialize(tag_name, markup, parse_context)
       end
 
       def render(context)
+        @source_dir = context.registers[:site].config['source']
```

Comment:
I think we don't need to store the source directory as an instance variable because it can change depending on the context. Instead, we can pass the context to `render_linkpreview`.

Reply:
Fixed as requested.